modules/javafx.web/src/test/java/test/javafx/scene/web/FileReaderTest.java

```diff
@@ -0,0 +1,252 @@
+        testLatch(latch);
+        submit(() -> {
+            assertEquals("File Content matches", "Hello Wo", getEngine().executeScript("window.result"));
+        });
+    }
+
+    @Test public void testreadAsText_slice_valid_parameters_7() {
+        loadContent(getScriptString("readAsText", ".slice(file.length, -100)", false));
+        testLatch(latch);
+        submit(() -> {
+            assertEquals("File Content matches", "", getEngine().executeScript("window.result"));
+        });
+    }
+
+    @Test public void testreadAsText_slice_single_parameter() {
+        loadContent(getScriptString("readAsText", ".slice(6)", false));
+        testLatch(latch);
+        submit(() -> {
+            assertEquals("File Content matches", "World", getEngine().executeScript("window.result"));
+        });
+    }
+
+    @Test public void testreadAsText_slice_negetive_parameter_1() {
+        loadContent(getScriptString("readAsText", ".slice(-3)", false));
+        testLatch(latch);
+        submit(() -> {
+            assertEquals("File Content matches", "rld", getEngine().executeScript("window.result"));
+        });
+    }
+
+    @Test public void testreadAsText_slice_negetive_parameter_2() {
+        loadContent(getScriptString("readAsText", ".slice(-3, -7)", false));
+        testLatch(latch);
+        submit(() -> {
+            assertEquals("File Content matches", "", getEngine().executeScript("window.result"));
+        });
+    }
+
+    @Test public void testreadAsBinaryString() throws FileNotFoundException, IOException {
+        String binaryFile[] = {new File("src/test/resources/test/html/BinaryFile.dat").getAbsolutePath()};
+        FileReaderShim.test_setChooseFiles(binaryFile);
+        loadContent(getScriptString("readAsBinaryString", "", false));
+        testLatch(latch);
+        FileInputStream in = new FileInputStream(binaryFile[0]);
+        final byte[] expectedBinaryData = in.readAllBytes();
+        submit(() -> {
+            final String obj = (String) getEngine().executeScript("window.result");
+            final byte[] binBytes = obj.getBytes();
+            for (int i = 0; i < expectedBinaryData.length; i++) {
+                assertEquals("File Content matches", expectedBinaryData[i], binBytes[i]);
+            }
+        });
+    }
+
+    @Test public void testreadAsArrayBuffer() throws FileNotFoundException, IOException {
+        loadContent(getScriptString("readAsArrayBuffer", "", false));
+        testLatch(latch);
+        FileInputStream in = new FileInputStream(fileList[0]);
+        final byte[] expectedArrayBuffer = in.readAllBytes();
+        submit(() -> {
+            final JSObject obj = (JSObject) getEngine().executeScript("new Uint8Array(window.result)");
+            assertEquals("Expected File Content matches", expectedArrayBuffer.length, 11);
+            assertEquals("File Content matches", obj.getMember("length"), 11);
```

Comment:
Above 2 assertion should be combined
`assertEquals(String.format("%s length must be equal in both Java & JavaScript", fileList) , expectedArrayBuffer.length, obj.getMember("length"));`